boards/beagle/beagleplay_cc1352/beagleplay_cc1352_defconfig

```diff
@@ -0,0 +1,21 @@
+#
+# Copyright (c) 2022 Jason Kridner, BeagleBoard.org Foundation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+CONFIG_GPIO=y
+CONFIG_PINCTRL=y
```

Comment:
Well, just doing a basic search, I can see that it is present in a lot of boards:
```
 rg "CONFIG_PINCTRL=y" boards -l | wc -l
296
```
But maybe it is not supposed to go there now? I can see some boards have it defined in the soc `Kconfig.defconfig` so maybe it is supposed to go there now?

Reply:
They do not follow best practices, need to be updated.

Reply:
> They do not follow best practices, need to be updated.

Ok, so where should Pinctrl enabled now?

Reply:
grep for `select PINCTRL` to see patterns

Reply:
> grep for `select PINCTRL` to see patterns

I can see that `espressif` soc do it in `Kconfig` and some drivers seem to do it, I guess. I think you are saying it should be done how `espressif` is doing? Since it is the only one using it, I am not quite sure.